var/spack/repos/builtin/packages/mesa/package.py

```diff
@@ -61,8 +61,12 @@ class Mesa(MesonPackage):
     depends_on("flex", type="build")
     depends_on("gettext", type="build")
     # Upperbound on 3.11 because distutils is used for checking py-mako
-    depends_on("python@3:3.11", type="build")
+    # See issue https://gitlab.freedesktop.org/mesa/mesa/-/issues/9943
+    # This was fixed with commit 92cdf6cb5f25f632350d8ca0d9451b0e87194ca3
+    # and incorporated into 23.3
+    depends_on("python@3:3.11", when="@:23.2", type="build")
```

Comment:
```suggestion
    # https://gitlab.freedesktop.org/mesa/mesa/-/commit/670a799ebff9a98daafccf49324c2a01311b0c41
    depends_on("python@:3.11", when="@:23.2", type="build")
```

You cannot combine forward and backward compat in a single depends_on statement.

But the backwards compat bound is not really necessary since we don't support Python 2 anyways.

Reply:
Also, it seems like we need a separate `depends_on` for versions `@23.3:`, otherwise those versions will not depend on python, at least not directly. I guess the dependency on python packages like `py-make` brings the dependency on `python` indirectly but it still maybe good to have the explicit dependency.